src/traits.rs

```diff
@@ -17,6 +17,13 @@ pub trait InputLength {
   fn input_len(&self) -> usize;
 }
 
+impl<T> InputLength for [T] {
```

Comment:
I'm not sure about it. Maybe crlf? I configured vscode to use lf always, but it somehow prefers crlf. I'll try to remove it from pull request.